test/DateExtensionTest.php

```diff
@@ -67,7 +68,7 @@ public function testDiffCanReturnTranslatableString($expected, $translated, $dat
         $translator = $this->getMockBuilder(TranslatorInterface::class)->getMock();
         $translator
             ->expects($this->once())
-            ->method('transChoice')
```

Comment:
We also need to keep a separate test using the legacy interface.

Reply:
Would you rather have 2 tests or some logic that would determine which interface to use?

Reply:
Well, the thing is, it is not the same logic. One uses `trans()` and the other one `transChoice`. Currently, one of the branches is untested.

Reply:
and that should be 2 tests. We want to cover both branches all the time (well, if some of them cannot be tested on some Symfony version due to not having the interface, the test might be skipped)

Reply:
Added one more test (and one more build configuration):

* Legacy interface is tested on 3.x, new one is skipped
* Both interfaces are tested on 4.x
* New interface is tested on 5.x, old one is skipped